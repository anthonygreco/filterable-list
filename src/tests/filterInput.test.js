import React from 'react'
import { render, Simulate } from 'react-testing-library'

import FilterInput from '../components/filterInput'

test('<FilterInput /> should produce a proper snapshot', async () => {
  const { container } = render(
    <FilterInput />
  )

  expect(container).toMatchSnapshot()
})

test('<FilterableInput /> should call onFilter prop method properly', async () => {
  const onFilter = jest.fn()
  const { container } = render(
    <FilterInput onFilter={onFilter} onClearFilter={() => {}} />
  )

  const filterRef = container.querySelector('input[type=text]')
  filterRef.value = 'a'
  Simulate.change(filterRef)

  expect(onFilter).toHaveBeenCalled()
})

test('<FilterableInput /> should call onClearFilter prop method properly', async () => {
  const onClearFilter = jest.fn()
  const { container } = render(
    <FilterInput onFilter={() => {}} onClearFilter={onClearFilter} />
  )

  const filterRef = container.querySelector('input[type=text]')
  filterRef.value = ''
  Simulate.change(filterRef)

  expect(onClearFilter).toHaveBeenCalled()
})
