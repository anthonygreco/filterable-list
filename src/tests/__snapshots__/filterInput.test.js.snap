// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterInput /> should produce a proper snapshot 1`] = `
<div>
  <div
    class="ui fluid icon left corner labeled input"
  >
    <div
      class="ui label label left corner"
    >
      <i
        aria-hidden="true"
        class="asterisk icon"
      />
    </div>
    <input
      type="text"
      placeholder="Filter..."
    />
    <i
      aria-hidden="true"
      class="search icon"
    />
  </div>
</div>
`;
